save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityHeader.kt

```diff
@@ -189,12 +197,73 @@ internal val headerMenu: FC<HeaderMenuProps> = FC { props ->
         }
     }
 
+    @Suppress("EMPTY_BLOCK_STRUCTURE_ERROR")
+    displayModal(
+        cvssCalculatorWindowOpenness.isOpen(),
+        "Base Score Calculator".t(),
+        bodyBuilder = {
+            cvssBaseScoreCalculator {
+                baseMetricsOfVulnerability = props.vulnerability.cosv.severity?.let { severityList ->
+                    val severityV3Count = severityList.count { it.type == SeverityType.CVSS_V3 }
+
+                    if (severityV3Count == 1) {
+                        val severityV3 = severityList.find { it.type == SeverityType.CVSS_V3 }!!
+                        CvssVectorV3(severityV3.score.parsingVectorToMap()).baseMetrics
+                    } else {
+                        undefined
+                    }
+                } ?: undefined
+
+                onCloseButton = { cvssCalculatorWindowOpenness.closeWindow() }
+
+                onCloseButtonPassed = { baseMetrics ->
+                    props.vulnerability.cosv.severity?.let { currentSeverityList ->
+                        val severityV2Count = currentSeverityList.count { it.type == SeverityType.CVSS_V2 }
+                        val severityList = currentSeverityList.toMutableList()
```

Comment:
There is no need to use MutableList here. It is better to use the `plus()` and `minus()` method of the `List`.